app/services/PaymentAPIService.scala

```diff
@@ -64,25 +64,56 @@ class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String) {
       .execute()
   }
 
-  def isDuplicatePaymentResponse(response: WSResponse): Boolean = {
+  def decodePaymentResponse(response: WSResponse): Option[PaymentApiError] = {
     decode[PaymentApiError](response.body).fold(
       failure => {
         val failureMessage = SafeLogger.LogMessage(
           s"Unable to decode PaymentAPIError: ${response.body}. Message is ${failure.getMessage}",
           "Unable to decode PaymentAPIError. See logs for details"
         )
         SafeLogger.error(failureMessage)
-        false
+        None
       },
-      paymentApiError => paymentApiError.error.errorName.contains("PAYMENT_ALREADY_DONE")
+      resp => {
+        Some(resp)
+      }
     )
   }
 
+  def logErrorResponse(paymentApiError: Option[PaymentApiError]): Unit = {
+    paymentApiError.foreach(err => {
+      if (err.error.errorName.contains("INSTRUMENT_DECLINED")) {
+        SafeLogger.info("Paypal payment failed with 'INSTRUMENT_DECLINED' response.")
+      } else {
+        val errorMessage = SafeLogger.LogMessage(
```

Comment:
You can use:
_scrub"string ${x.name}"_ 

instead of:
_s"string {x.name}"_

This will build the LogMessage and deal with the scrubbing for you:
https://github.com/guardian/support-frontend/blob/ef4091678869f9fc71f71b7bc9845b49a192ded1/app/monitoring/Logging.scala#L25

Reply:
So this whole block would become:

_SafeLogger.error(scrub"Paypal payment failed due to ${err.error.errorName} error. Full message: ${err.error.message}")_

Reply:
nice!